### DISCUSSION TITLE
File.ExistsAsync missing from the async File methods.

### DISCUSSION THREAD

**Discussion by CryoMyst** [dotnet/runtime#67587] [General]
Why is this missing from the `File` methods?

Seems useful for networks shares or slow raid arrays on HDDs that have spun down.

**Comment by stephentoub** [comment]
How would you implement it?

**Comment by teo-tsirpanis** [comment]
Hello @CryoMyst, unfortunately there is no OS support for asynchronously checking whether a file exists. Perhaps it could be added on Linux with `io_uring` but exposing new platform-specific async I/O APIs is a discussion that would better come after `io_uring` itself.

**Reply by huoyaoyuan** [reply]
As you can see, there's no `OpenAsync` `DeleteAsync` etc either. Currently async *filesystem* (instead of async IO) is not supported for most platforms.
